plugins/gitlab-catalog-backend/src/GitlabUserProcessor.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import { Entity, isUserEntity } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+import type {
+  CatalogProcessor,
+  CatalogProcessorEmit,
+} from '@backstage/plugin-catalog-node';
+import { ExpandedUserSchema, Gitlab } from '@gitbeaker/rest';
```

Comment:
decided on whether it was worth importing another library for the gitlab client even if it wasn't an official client, and this seemed like a good choice because its already included in backstage as a dependency among other known oss repos